Sources/Quick/Example.swift

```diff
@@ -10,6 +10,11 @@ public class _ExampleBase: NSObject {}
 // swiftlint:enable type_name
 #endif
 
+public enum QuickError: Error {
+    case stopSilently
+    case stop(_ description: String, file: FileString = #file, line: UInt = #line)
+}
```

Comment:
I don't think this should be called `QuickError`. Furthermore, I think this should be a struct, not an enum.

The `Error` part implies that this is, you know, an error. That something went catastrophically wrong. What we aim to do with this is provide a mechanism for stopping test execution, and (possibly, in most cases) report a test failure. The fact that this type conforms to `Error` is an artifact of the mechanism we use to stop test execution.

The `Quick` part implies that this is internal to Quick, not something that should be used by people using Quick.

Another reason I think this is poorly named is that only the case names give any clue as to what this is for.

Secondly, I think this should be a struct, not an enum. I don't think the `stopSilently` and `stop` cases need to be two variants of what is mostly the same thing. At best, this could be a boolean parameter on the initializer. However, my opinion on this matter is much more weakly held, and I could definitely see keeping this as an enum.

So, my proposed change:

```swift
public struct StopTest: Error {
    public let description: String
    public let reportError: Bool
    public let file: FileString
    public let line: UInt

    public init(_ description: String, reportError: Bool = true, file: FileString = #file, line: UInt = #line) {
        self.description = description
        self.reportError = reportError
        self.file = file
        self.line = line
    }

    public static var silently: StopTest {
        StopTest("", reportError: false)
    }
}
```

And usage would look like:

```swift
guard [1, 2].isEmpty else { throw StopTest("Expected error to be empty, but it wasn't") }
```

Reply:
I took this approach and made some small changes. The big one is that I'm not exposing `reportError` to the public initializer since it's mutually exclusive from the other properties.  That's why I initially leaned towards an enum.

The reason I think a struct really makes sense here is the callsite for developers.  `StopTest("Things went wrong")` is more intuitive than something like `StopTest.loudly("Things went wrong")`.